controllers/policyserver_controller_test.go

```diff
@@ -70,7 +93,44 @@ var _ = Describe("PolicyServer controller", func() {
 			)
 		})
 
+		It("should not delete its managed resources until all the scheduled policies are gone", func() {
+			By("having still policies pending deletion")
+			Expect(
+				getTestClusterAdmissionPolicy(policyName),
+			).To(
+				And(
+					HaveField("DeletionTimestamp", Not(BeNil())),
+					HaveField("Finalizers", Not(ContainElement(constants.KubewardenFinalizer))),
+					HaveField("Finalizers", ContainElement(IntegrationTestsFinalizer)),
+				),
+			)
+
+			Expect(
+				getTestPolicyServerService(policyServerName),
+			).To(
+				HaveField("DeletionTimestamp", BeNil()),
+			)
```

Comment:
Looking back at that, we do not add finalizers to the Service object associated with the Policy Server (we don't need to). Because of that, we either will find the Service or not find it.

Can you rewrite this assertion to be:

```golang
Eventually(func(g Gomega) error {
	_, err := getTestPolicyServerService(policyServerName)
	return err
}, timeout, pollInterval).Should(Succeed())
```